assembly/src/tests.rs

```diff
@@ -1354,6 +1354,31 @@ end";
     Ok(())
 }
 
+#[test]
+fn program_with_proc_locals_fail() -> TestResult {
+    let mut context = TestContext::default();
+    let source = source_file!(
+        "\
+        proc.foo \
+            loc_store.0 \
+            add \
+            loc_load.0 \
+            mul \
+        end \
```

Comment:
Do we already have a check that something like this would also be an error?:
```
proc.foo.1
    loc_store.0
    add
    loc_load.1
    mul
end
```

Reply:
Yes, we are doing this check [here](https://github.com/0xPolygonMiden/miden-vm/blob/46aca51b9776632316cbdd061aced6991640251b/assembly/src/assembler/instruction/mem_ops.rs#L127) in `validate_param` function. So in that example it will return error with message "parameter value must be greater than or equal to 0 and less than or equal to 0, but was 1".